### PR TITLE
Added documentation for Presto on Spark

### PR DESCRIPTION
**Pull request by Ravion**
Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.

```
== RELEASE NOTES ==

General Changes
* ...
* ...

Hive Changes
* ...
* ...
```

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by Ravion** [comment]
@arhimondr  Sorry that I had to close the other PR. Opened a new one with all changes advised are being added. Below were your comments:

1) Could you please add a paragrapth describing this catalog structure (or point it to the section from the Presto documentation https://prestodb.io/docs/current/installation/deployment.html#catalog-properties
DONE

2) Please add a paragraph explaining this. We need to set spark.task.cpus equal to the --executor-cores to make sure only a single Presto on Spark task is run in a single Spark executor. We introduced this limitation to avoid duplicating broadcasted hash table for every task. We may optimize this in the future and remove this limitation. Also the task.concurrency, task.max-worker-threads (and possibly task.writer-count) should be adjusted to keep all executor cores busy. For example if --executor-cores 4 the task.concurrency, task.max-worker-threads must be set to 4. You can actually include this properties in the config.properties example.
DONE

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+an option to scale the resource management using Spark, as it doesn'f offer
+one currently by itself. This way Presto can be executed as a stand alone 
+application on Spark.
+
+Steps
+-----
+
+Download the Presto Spark package tarball, :maven_download:`spark-package` 
+and the Presto Spark laucher, :maven_download:`spark-launcher`. Keep both the
+files at, say, *example* directory. We assume here a two node Spark cluster
```
Comment (arhimondr) [inline presto-docs/src/main/sphinx/installation/spark.rst:15]: > with 2 cores each, thus giving us total four cores

If each spark workers has `2` cores then `task.concurrency`/`task.max-worker-threads`/`task.writer-count` and `spark.task.cpus`/`--executor-cores` must be set to `2`.

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+with 2 cores each, thus giving us total four cores.
+
+The following is an example ``config.properties``:
+
+.. code-block:: none
+
+    query.hash-partition-count=10
```
Comment (arhimondr) [inline presto-docs/src/main/sphinx/installation/spark.rst:22]: This is not required. Let's not add it.

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    redistribute-writes=false
```
Comment (arhimondr) [inline presto-docs/src/main/sphinx/installation/spark.rst:23]: ditto

**Comment by arhimondr** [comment]
@aweisberg @wenleix could you please have a look?

**Comment by Ravion** [comment]
@arhimondr @wenleix @aweisberg 

Amended latest review comments by @arhimondr . Add screenshots as well. Kindly review. Thanks

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
+=========================
+Executing Presto on Spark
+=========================
+
+When there is a need to execute long running queries, Presto offers
```
Comment (wenleix) [inline presto-docs/src/main/sphinx/installation/spark.rst:5]: nit: long running or memory-intensive queries

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+an option to scale the resource management using Spark, as it doesn'f offer
```
Comment (wenleix) [inline presto-docs/src/main/sphinx/installation/spark.rst:6]: nit: 
- scale the resource management using Spark -> scale the resource management *by* using Spark
- as it doesn'f offer -> as it doesn't offer

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+one currently by itself. This way Presto can be executed as a stand alone
```
Comment (wenleix) [inline presto-docs/src/main/sphinx/installation/spark.rst:7]: nit: stand alone -> standalone

File: presto-docs/src/main/sphinx/installation/spark.rst (hunk 1 above)
Comment (wenleix) [reply presto-docs/src/main/sphinx/installation/spark.rst:15]: wha's this "example" directory?  didn't see it get referred in command and other part of the doc?

File: presto-docs/src/main/sphinx/installation/spark.rst [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+The following is an example ``config.properties``:
+
+.. code-block:: none
+
+    task.concurrency=4
+    task.max-worker-threads=4
+    task.writer-count=4
+     
+The details about properties are available at :doc:`/admin/properties`.
+Note that ``task.concurrency``, ``task.writer-count`` and
```
Comment (wenleix) [inline presto-docs/src/main/sphinx/installation/spark.rst:27]: @arhimondr : I remember you mentioned the number of threads should be slightly higher than the number of physical cores to get the best resource utilization?

File: presto-docs/src/main/sphinx/installation/spark.rst (hunk 6 above)
Comment (Ravion) [reply presto-docs/src/main/sphinx/installation/spark.rst:7]: Changed, thanks

File: presto-docs/src/main/sphinx/installation/spark.rst (hunk 5 above)
Comment (Ravion) [reply presto-docs/src/main/sphinx/installation/spark.rst:6]: changed , thanks

File: presto-docs/src/main/sphinx/installation/spark.rst (hunk 7 above)
Comment (Ravion) [reply presto-docs/src/main/sphinx/installation/spark.rst:27]: Incorporated changes by @wenleix ..Please review, thanks

**Comment by arhimondr** [comment]
@Ravion 

Could you please rebase it?


Here is a nice explanation of how rebase work: https://git-scm.com/book/en/v2/Git-Branching-Rebasing

**Comment by Ravion** [comment]
@Andrill Like before I faced some issue when rebasing with so many unwanted files. Updated all review coments so far and submitted #14785. Kindly review. 

Please let me ask offline on some issues I faced during rebase. Thanks

**Pull request closed without merging**